packages/twenty-front/src/modules/object-record/record-filter/utils/isRecordMatchingFilter.ts

```diff
@@ -268,6 +269,17 @@ export const isRecordMatchingFilter = ({
           })
         );
       }
+      case FieldMetadataType.Emails: {
+        const emailsFilter = filterValue as EmailsFilter;
+
+        return (
+          emailsFilter.primaryEmail === undefined ||
```

Comment:
why is it a match in this case (emailsFilter.primaryEmail === undefined) ?

Reply:
Hello @ijreilly 👋 
We were following the same approach as the Actor field, but as we are also going to include additional fields, we will refactor this check.
And, about this, on the line you mentioned [here](https://github.com/twentyhq/twenty/pull/6689#discussion_r1724981560), the `additionalLinks` are not being used, the code uses the URL and the label of the primary link, should we include the `additionalEmails`  or keep following the Links field type?

Reply:
@gitstart-twenty it seems to be a flow in the links implementation. The expected behaviour is that when filtering, the additional emails are taken into account as well

Reply:
Hello @ijreilly , we just remembered that [Charles had mentioned](https://discord.com/channels/1130383047699738754/1163747555461570650/1273974123458068512) that the filter would only work on primary email, I believe this is the reason why the links do not use the primary link

Reply:
thanks, my bad then !